### ISSUE TITLE
CWRC MODS to DC transform

### ISSUE THREAD

**Issue by jefferya** [cwrc/islandora_cwrc_xslt_library#1]
@brundin @SusanBrown 

Problems identified the the LOC MODS to DC transform bundled with Islandora (detailed in [1]) are affecting:
(a) generated DC and resulting view
(b) search result items(with author and date pulled from the DC record)
(b) collection member item details (author and date pulled from the DC record)

This repo is meant to work in conjunction with islandora_xslt_paths module to allow the registration of a custom XSLT with a given form where the custom XSLT will reside in this repository (with this repo acting as a Drupal library).

@brundin @SusanBrown Can the needed customizations to the MODS to DC transform be added to the XSLT in this repo?

I'm adding a drush script to islandora_xslt_paths that will aid in applying the custom transform to existing material.

[1] https://groups.google.com/d/msg/islandora/Ot6N0eIvPjc/48ttrqnHAQAJ

**Comment by brundin** [comment]
@jefferya @SusanBrown 

I'm probably missing something obvious (and let me know if I am), but it appears that you have already added the needed customizations to the MODS to DC transform to the XSLT in this repo:

https://github.com/cwrc/islandora_cwrc_xslt_library/blob/master/xslt/mods_to_dc.xsl

Looks good!

**Comment by jefferya** [comment]
@brundin @SusanBrown This means I fatefully chose one of your test files as opposed to the original LOC file.

**Comment by brundin** [comment]
@jefferya @SusanBrown Ah, I see.

**Comment by jefferya** [comment]
@brundin @SusanBrown 

A related question. Are both "author" and "Author" valid roles in MODS (i.e., both lowercase and uppercase)? If yes, then we should add this to the custom mods_to_dc in this repo. I'm inferring "creator" and "Creator" follow the same rule, correct?

**Comment by brundin** [comment]
@jefferya @SusanBrown 

Technically, any string value is valid for the text node in the XPath 

/mods:mods/mods:name/mods:role/mods:roleTerm[author='marcrealtor'][type='text']/text()

The MODS schema does not enumerate a list of prescribed values for this text node.  So "XYZ" is a valid value in terms of the MODS schema for the <roleTerm> element text node.  This is also true for the @type="code" values as well.

However, the Library of Congress' MARC Code List for Relators: Term Sequence explicitly lists all of the @type="text" values as having initial-letter capitalization for the first word, so "Author", "Creator", "Editor", etc.  See

https://www.loc.gov/marc/relators/relaterm.html

The corresponding code value is given in square brackets after each term.  The MARC Code List for Relators: Code Sequence is listed here, with the corresponding term for each code following after the code:

https://www.loc.gov/marc/relators/relacode.html

So, "aut" ("Author"), "cre" ("Creator"), "edt" ("Editor"), etc.

So if we are being rigorously correct and authoritative, we should probably just have conditional checks for the initial-letter capitalization for "Author" and "Creator", and not "author" and "creator" as well, as that is how the Library of Congress defines them in their prescribed enumerated list.  If users are using "author" and "creator", they are not following the Library of Congress' defined MARC Code List for Relators: Term Sequence list of values.

**Comment by ghost** [comment]
So are these the XSLTs that were supposed to fix the problems with authors’ names not showing if they were not classes as creators but contributors? Or have it got it wrong?

**Comment by jefferya** [comment]
Yes, in the context MODS to DC transform and the resulting DC display.
